lib/eclipse/EclipseState/Schedule/MSW/Segment.cpp

```diff
@@ -109,6 +109,19 @@ namespace Opm {
         m_volume = volume_in;
     }
 
+    const std::vector<int>& Segment::inletSegments() const {
+        return m_inlet_segments;
+    }
+
+    void Segment::clearInletSegments() {
+        // in case the segment structure changes during the simulation
```

Comment:
I do not like this; preferably the variable stays constant for the entire simulation - and then no `clearXxx( )` method is required. In the case an immutable segment is not feasible we should perform the clearing exactly when and where it is needed - this "mutate the state just to certain" makes it more difficult to reason about the code.

Reply:
I do not know.  To me, one thing can come later while extremely difficult about multisegment well is that, we introduce a multisegment well, we run for some period, then we change some of the segments changing the segment structure, then we run  for some other period.  To change the segment structure, you only input the changed or the new ones.  `clearInletSegments()` is to consider that.  I can remove this function for now for sure without breaking anything.  I have to admit that I am not familiar with `DynamicState`. If you think with my case, each instance will be immutable with `DynamicState`, I will remove this function.

Reply:
> I can remove this function for now for sure without breaking anything.

Do that.